src/adapter/src/coord.rs

```diff
@@ -1428,12 +1428,11 @@ impl Coordinator {
                 }
             };
 
-            // All message processing functions trace. Start a parent span for them to make
-            // it easy to find slow messages.
-            let span = span!(Level::DEBUG, "coordinator message processing");
-            let _enter = span.enter();
-
-            self.handle_message(msg).await;
+            self.handle_message(msg)
+                // All message processing functions trace. Start a parent span for them to make
+                // it easy to find slow messages.
+                .instrument(span!(Level::DEBUG, "coordinator message processing"))
+                .await;
```

Comment:
@mjibson did you do this purposely because you wanted the wall time here?

Reply:
The enter and exit each time we poll will be shorter, but the start and end time of the span will be the wall time!